public/js/clients/firefox/commands.js

```diff
@@ -79,6 +83,12 @@ function removeBreakpoint(breakpointId) {
   return bpClient.remove();
 }
 
+function setBreakpointCondition(breakpointId, location, condition, noSliding) {
+  const bpClient = bpClients[breakpointId];
+  return bpClient.setCondition(threadClient, condition, noSliding)
+    .then((res) => onNewBreakpoint(location, [{}, res]))
```

Comment:
You should probably remove the old breakpoint client from `bpClients` here before we add a new one.